.github/workflows/buildAndTest.yaml

```diff
@@ -0,0 +1,24 @@
+name: Build And Test
+
+on: [push, pull_request]
+
+jobs:
+  buildAndTest:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v1
+      with:
+        java-version: 8
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build
+      run: ./mvnw compile
```

Comment:
You can add the `-q` flag which is useful for suppressing extra log messages (like the ones showing what % download is reached).

Reply:
For equivalence with what with have in `.travis.yaml`, this line would be:
```
./mvnw -T 1.5C install -DskipTests=true -Dmaven.javadoc.skip=true -Dspring.profiles.active=spring -P full-checkstyle -B -V
```

Reply:
Tried adding the -q flag, but it suppressed too much output - now it just has DEBUG logs, which aren't super useful without context.

I added a few of the same flags from the `travis.yaml` file - some of which we don't want though (like `skipTests!`)

I collapsed it into just running `test`, which should obviously catch any compile errors too.